modules/xfem/src/efa/EFAEdge.C

```diff
@@ -0,0 +1,419 @@
+      break;
+    }
+  }
+}
+
+void
+EFAEdge::copyIntersection(const EFAEdge & other, unsigned int from_node_id)
+{
+  _embedded_nodes.clear();
+  _intersection_x.clear();
+  _embedded_nodes = other._embedded_nodes;
+  if (from_node_id == 0)
+    _intersection_x = other._intersection_x;
+  else if (from_node_id == 1)
+  {
+    for (unsigned int i = 0; i < other.numEmbeddedNodes(); ++i)
+      _intersection_x.push_back(1.0 - other._intersection_x[i]);
+  }
+  else
+    EFAError("from_node_id out of bounds");
+  if (_embedded_nodes.size() != _intersection_x.size())
+    EFAError("in copyIntersection num emb_nodes must == num of inters_x");
+}
+
+EFANode *
+EFAEdge::getNode(unsigned int index) const
+{
+  if (index == 0)
+    return _edge_node1;
+  else if (index == 1)
+    return _edge_node2;
+  else
+    EFAError("In getNode index out of bounds");
+}
+
+void
+EFAEdge::reverseNodes()
+{
+  EFANode* tmp = _edge_node1;
+  _edge_node1 = _edge_node2;
+  _edge_node2 = tmp;
+  for (unsigned int i = 0; i < _embedded_nodes.size(); ++i)
+    _intersection_x[i] = 1.0 - _intersection_x[i];
+}
+
+bool
+EFAEdge::hasIntersection() const
+{
+  return (_embedded_nodes.size() > 0);
+}
+
+bool
+EFAEdge::hasIntersectionAtPosition(double position, EFANode * from_node) const
+{
+  double tol = 1.e-4;
+  bool has_int = false;
+  if (hasIntersection())
+  {
+    double tmp_intersection_x = -1.0;
+    if (from_node == _edge_node1)
+      tmp_intersection_x = position;
+    else if (from_node == _edge_node2)
+      tmp_intersection_x = 1.0 - position;
+    else
+      EFAError("In hasIntersectionAtPosition from_node does not exist on edge");
+
+    for (unsigned int i = 0; i < _embedded_nodes.size(); ++i)
+    {
+      if (std::abs(tmp_intersection_x - _intersection_x[i]) < tol)
+      {
+        has_int = true;
+        break;
+      }
+    }
+  }
+  return has_int;
+}
+
+double
+EFAEdge::getIntersection(unsigned int emb_id, EFANode * from_node) const
+{
+  if (from_node == _edge_node1)
+    return _intersection_x[emb_id];
+  else if (from_node == _edge_node2)
+    return 1.0 - _intersection_x[emb_id];
+  else
+    EFAError("In getIntersection node not in edge");
+}
+
+double
+EFAEdge::distanceFromNode1(EFANode * node) const
+{
+  double xi = -100.0;
+  if (_edge_node1 == node)
+    xi = 0.0;
+  else if (_edge_node2 == node)
+    xi = 1.0;
+  else if (isEmbeddedNode(node))
+  {
+    unsigned int embedded_node_id = getEmbeddedNodeIndex(node);
+    xi = _intersection_x[embedded_node_id];
+  }
+  else
+    EFAError("the given node is not found in the current edge");
+  return xi;
+}
+
+bool
+EFAEdge::isEmbeddedNode(const EFANode * node) const
+{
+  bool is_emb = false;
+  for (unsigned int i = 0; i < _embedded_nodes.size(); ++i)
+  {
+    if (_embedded_nodes[i] == node)
+    {
+      is_emb = true;
+      break;
+    }
+  }
+  return is_emb;
+}
+
+unsigned int
+EFAEdge::getEmbeddedNodeIndex(EFANode * node) const
+{
+  unsigned int index = 99999;
```

Comment:
Yes, please change this. I went through all of MOOSE and took out all the other random number of 9 initializations. We don't need this as Daniel mentioned. You can verify this with "make sa" yourself.